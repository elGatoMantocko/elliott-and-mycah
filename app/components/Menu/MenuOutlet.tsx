import { Camera as CameraIcon, Home as HomeIcon, Menu as MenuIcon } from '@mui/icons-material';
import {
  AppBar,
  Box,
  Container,
  Drawer,
  Fab,
  Hidden,
  List,
  ListItemIcon,
  ListItemText,
  Toolbar,
} from '@mui/material';
import React, { useState } from 'react';
import { Outlet } from 'react-router';

import { ScriptTypography } from '../ScriptTypography';
import { ActiveLink } from './ActiveLink';
import { ListItemLink } from './ListItemLink';

export const MenuOutlet = () => {
  const [drawerOpen, setDrawerOpen] = useState(false);
  const closeDrawer = () => setDrawerOpen(false);

  return (
    <>
      <Hidden mdDown>
        <AppBar position="static" color="transparent" elevation={0}>
          <Container>
            <Toolbar>
              <Box>
                <ActiveLink to="/us">
                  <ScriptTypography>About us</ScriptTypography>
                </ActiveLink>
              </Box>
              <Box ml={6}>
                <ActiveLink to="/pictures">
                  <ScriptTypography>Pictures</ScriptTypography>
                </ActiveLink>
              </Box>
              <Box ml={6}>
                <ActiveLink to="/wedding">
                  <ScriptTypography>Wedding</ScriptTypography>
                </ActiveLink>
              </Box>
            </Toolbar>
          </Container>
        </AppBar>
      </Hidden>
      <Hidden mdUp>
        <Box position="fixed" bottom="0" right="0" margin={3} zIndex={1000}>
          <Fab color="secondary" onClick={() => setDrawerOpen(true)}>
            <MenuIcon />
          </Fab>
        </Box>
        <Drawer anchor="bottom" open={drawerOpen} onClose={closeDrawer}>
          <List>
            <ListItemLink to="/us" onClick={closeDrawer}>
              <ListItemIcon>
                <HomeIcon />
              </ListItemIcon>
              <ListItemText primary="About us" />
            </ListItemLink>
            <ListItemLink to="/pictures" onClick={closeDrawer}>
              <ListItemIcon>
                <CameraIcon />
              </ListItemIcon>
              <ListItemText primary="Pictures" />
            </ListItemLink>
            <ListItemLink to="/wedding" onClick={closeDrawer}>
              <ListItemIcon>
                <HomeIcon />
              </ListItemIcon>
              <ListItemText primary="Wedding" />
            </ListItemLink>
          </List>
        </Drawer>
      </Hidden>

      <Outlet />
    </>
  );
};
