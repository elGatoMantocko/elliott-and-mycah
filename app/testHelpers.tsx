import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { match } from 'css-mediaquery';
import React from 'react';
import { MemoryRouter, Route, Routes, useLocation } from 'react-router';

type TestLocations = 'route-location';

/**
 * The element testid used to show the location.
 */
export const TestRouteLocation: TestLocations = 'route-location';

interface TestLocationProps {
  testId?: string;
}

/**
 * Component that exposes the location path and search to the dom
 * @param root0 props
 * @param root0.testId test id passed to the inner location element
 * @returns element
 */
const TestLocation = ({ testId }: TestLocationProps) => {
  const location = useLocation();
  return (
    <div data-testid={testId ?? TestRouteLocation}>{`${location.pathname}${location.search}`}</div>
  );
};

interface RouterRenderOptions extends RenderOptions {
  locationTestId?: string;
}

/**
 * Use to render UI with a react-router with the ability to search for a `route-location` test id.
 * @example
 * ```typescriptreact
 * const el = renderWithRouter(<MyComponent />);
 * const locationElementData = await el.findByTestId(TestRouteLocation);
 * ```
 * @param ui to render in the test
 * @param options passed to testing-library
 * @param options.locationTestId passed to the `<TestLocation />` element
 * @returns render result
 */
export const renderWithRouter = (
  ui: React.ReactElement,
  { locationTestId, ...renderOptions }: RouterRenderOptions = {},
): RenderResult => {
  return render(
    <MemoryRouter>
      {ui}
      <Routes>
        <Route path="*" element={<TestLocation testId={locationTestId} />} />
      </Routes>
    </MemoryRouter>,
    renderOptions,
  );
};

// eslint-disable-next-line @typescript-eslint/no-empty-function
const noop = () => {};

/**
 * Function to create a match media for mocking media queries.
 * @param width to mock in the jsdom
 * @returns matchMedia options
 */
export function createMatchMedia(width: number): (query: string) => MediaQueryList {
  return (query: string) =>
    ({
      matches: match(query, { width }),
      addListener: noop,
      removeListener: noop,
    }) as unknown as MediaQueryList;
}
