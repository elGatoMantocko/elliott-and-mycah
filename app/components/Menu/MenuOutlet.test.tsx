import React from 'react';

import { renderWithRouter } from '../../testHelpers';
import { MenuOutlet } from './MenuOutlet';

it('should render a <Menu />', async () => {
  const el = renderWithRouter(
    <div data-testid="menu">
      <MenuOutlet />
    </div>,
  );
  expect(await el.findByTestId('menu')).toMatchSnapshot();
});
