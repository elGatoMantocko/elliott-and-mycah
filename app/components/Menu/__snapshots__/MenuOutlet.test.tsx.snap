// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a <Menu /> 1`] = `
<div
  data-testid="menu"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-qztcla-MuiTypography-root-MuiLink-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vt3dvp-MuiTypography-root"
            >
              About us
            </p>
          </a>
        </div>
        <div
          class="MuiBox-root css-12x7e8k"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-qztcla-MuiTypography-root-MuiLink-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vt3dvp-MuiTypography-root"
            >
              Pictures
            </p>
          </a>
        </div>
        <div
          class="MuiBox-root css-12x7e8k"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-qztcla-MuiTypography-root-MuiLink-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vt3dvp-MuiTypography-root"
            >
              Wedding
            </p>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
